src/group_theory/sylow.lean

```diff
@@ -292,6 +292,31 @@ lemma card_sylow_dvd_index [fact p.prime] [fintype (sylow p G)] (P : sylow p G)
   card (sylow p G) ∣ P.1.index :=
 ((congr_arg _ (card_sylow_eq_index_normalizer P)).mp dvd_rfl).trans (index_dvd_of_le le_normalizer)
 
+lemma not_dvd_index_sylow' [hp : fact p.prime] (P : sylow p G) [P.1.normal] (hP : P.1.index ≠ 0) :
```

Comment:
Is the index being non-zero the Official Mathlib Way of saying that a subgroup has finite index? This feels a bit like saying a set is finite by including a hyopthesis `(h : nat.card s ≠ 0)`

How about:
```suggestion
lemma not_dvd_index_sylow' [hp : fact p.prime] (P : sylow p G) [P.1.normal] [fintype $ G ⧸ (P : subgroup G)] :
```

Reply:
In addition isn't `(P : subgroup G)` preferable to `P.1`?

Reply:
> Is the index being non-zero the Official Mathlib Way of saying that a subgroup has finite index? This feels a bit like saying a set is finite by including a hyopthesis `(h : nat.card s ≠ 0)`

I don't know if there is an "Official Mathlib Way", since I'm the primary group theory contributor right now. One example not written by me is `commensurable.lean`, which uses `relindex H K ≠ 0` rather than `fintype $ K ⧸ (H.subgroup_of K)`. Personally, I prefer `index ≠ 0`, because `index.lean` has good API for transferring "finite (rel)index" between subgroups. Also, if the subgroup is not normal, then stating finite index as `fintype $ G ⧸ H` feels a bit weird, since the space of left cosets is not natural (e.g., why not use right cosets instead?).

Reply:
Regarding using `(P : subgroup G)`, I think it is preferable, and `P.1` is just a bad habit that I've gotten into to save characters. But it would be nice if there was a way to avoid writing `(P : subgroup G)` throughout the proofs.

Reply:
I pushed a commit that uses `P : subgroup G` throughout the file.

Reply:
What do you think about having a `finite_index` typeclass, which in turn would automatically give `fintype (G ⧸ H)`? I'll be away the next few days, but when I get back I could add some basic API to the end of `index.lean`. Also, such a typeclass would get rid of this annoying @: https://github.com/leanprover-community/mathlib/blob/ffad43d259e5eb5084c2b76ace1ffa417aec33d5/src/group_theory/schreier.lean#L139

Reply:
Ok, I'll throw together a PR when I get back.